Work

ritasumtsova opened the pull request:
1. Fixed the comments;
2. Added Redux stuff for the DepartmentForm component;
3. Added API.addDepartment() to the Redux;
4. Added submitHandler to the modal in case user adds a department.

Code is not optimized good just covered add department modal form submit to check it works.

ritasumtsova committed:
worked on redux

ritasumtsova committed:
regular commit

ritasumtsova committed:
worked on modal forms redux

ritasumtsova committed:
working on redux submit forms

ritasumtsova committed:
replaced modals name with enum

ritasumtsova committed:
fixed the comments, added redux stuff for the DepartmentForm comp, added submitHandler to the modal in case user adds a department, added API.addDepartment() to the Redux

ritasumtsova committed:
removed comments

didevgen commented on src/components/DepartmentForm/DepartmentForm.tsx:
```diff
@@ -1,13 +1,34 @@
+        placeholder="Description"
+        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+          dispatch(changeDepartmentDescripton(e.target.value));
+        }}
```
you don't need to update the store each time you are typing. Do that on submit instead

didevgen commented on src/components/ModalWindow/ModalWindow.tsx:
```diff
@@ -7,27 +7,38 @@ import {
-    return name === currentModalContent?.name;
-  });
-
+  const submitHandler = () => {
```
wrap in useCallback

didevgen commented on src/store/reducers/formsReducer.ts:
```diff
@@ -0,0 +1,29 @@
+  description: ''
+};
+
+const formsReducer = (state: FormInitState = initState, action: ActionType) => {
```
it is better to rely on useReducer from react to handle such logic. (you will not mutate a global state in a such scenario)
But better to use formik or similar form-control manager

ritasumtsova committed:
removed forms redux stuff, added sumbit handlers to the departmentsForm and employeeForm, added addEmployee() redux stuff

ritasumtsova committed:
added success and failure modals

ritasumtsova committed:
replaced succes & failure modals with the one message modal

ritasumtsova committed:
added modals content enums

ritasumtsova commented:
1. Added a MessageModal component
2. Added content modals enum
3. Added dispatch open nested modal in the modal forms
4. Added useRef to dispatch the whole modal forms
5. Removed forms redux stuff

ritasumtsova committed:
fixed add employee request

ritasumtsova committed:
fixed a little departments reducer

didevgen commented on src/components/DepartmentForm/DepartmentForm.tsx:
```diff
@@ -1,13 +1,63 @@
 import './DepartmentForm.scss';
 
 const DepartmentForm: React.FC = () => {
+  const nameRef = useRef<HTMLInputElement | null>(null);
+  const descritpionRef = useRef<HTMLInputElement | null>(null);
```
o_O
all form elements should have value and onChange properties. Use state to manipulate that

didevgen commented on src/components/EmployeeForm/EmployeeForm.tsx:
```diff
@@ -1,15 +1,80 @@
+}
+
+const EmployeeForm: React.FC<EmployeeFormProps> = ({ id }) => {
+  const firstName = useRef<HTMLInputElement | null>(null);
+  const lastName = useRef<HTMLInputElement | null>(null);
+  const userName = useRef<HTMLInputElement | null>(null);
+  const email = useRef<HTMLInputElement | null>(null);
+  const dispatch = useDispatch();
+  const thunkDispatch = useDispatch<AppThunkDispatch>();
+
+  const submitHandler = useCallback(
+    (e: React.FormEvent<HTMLFormElement>) => {
+      e.preventDefault();
+
+      const data: Employee = {
+        _id: id,
+        firstName: firstName!.current!.value,
+        lastName: lastName!.current!.value,
+        userName: userName!.current!.value,
+        email: email!.current!.value
+      };
```
same here

didevgen replied:
But here useReducer instead of useState

ritasumtsova committed:
replaced the refs with the useState in the DepartmentForm & EmployeeForm

ritasumtsova committed:
added focus on first name input to the EmployeeForm comp

didevgen commented on src/components/DepartmentForm/DepartmentForm.tsx:
```diff
@@ -1,13 +1,73 @@
+    }, [thunkDispatch]
+  );
+
+  useEffect(() => {
+    nameRef.current?.focus();
+  })
```
you need to specify dependencies always

ritasumtsova replied:
my bad, fixed that

ritasumtsova committed:
added the dependencies to the useEffect input focus
